api/v1alpha1/openapi.yaml

```diff
@@ -1993,20 +1993,29 @@ components:
           description: 'Current state of the device.'
           items:
             $ref: '#/components/schemas/Condition'
-        systemInfo:
+        systemInfo: # deprecated
           $ref: '#/components/schemas/DeviceSystemInfo'
-        containers:
+        containers: # deprecated
           type: array
           description: Statuses of containers in the device.
           items:
             $ref: '#/components/schemas/ContainerStatus'
-        systemdUnits:
+        systemdUnits: # deprecated
           type: array
           description: 'Current state of systemd units on the device.'
           items:
             $ref: '#/components/schemas/DeviceSystemdUnitStatus'
+        workload:
```

Comment:
I was thinking virtual machines might not fit the application model. Plus other things we might not of thought of. But I'm open to moving it to root level.

Reply:
Yeah, I agree workloads are broader and cover non-applications, too. system services probably also fall into that bucket. More importantly, though, applications are the a world view from the user down, what performs the tasks they want to get done. Workload is the view from the system up, it’s what puts load onto system resources. VMs can be both: a workload or application (appliance).

To sum up, I agree there’d potentially be other workloads. Would they be summarized under the ApplicationStatusSummary, too?

Somehow I’m wondering whether the schema should be flatter, i.e. whether we need the workload/system/config buckets at the top.

Reply:
If we keep this structure the highest level object would summarize its children. So WorkloadSummary would summarize all workload types.

ApplicationSummary would summarize all applications. High level are all workloads working without error. Then each level provides more granular details.

If every level is on the root. You might get simpler queries but lose the relationship or need to reconstruct it.

I think if we keep the api at most 2 levels it should be fine but we could benchmark to understand the trade offs on read/write?

Reply:
> Somehow I’m wondering whether the schema should be flatter, i.e. whether we need the workload/system/config buckets at the top.

If you can outline the advantage you see for this change it can help me weigh that. My general goal is to provide a simple dependency tree. So the status can be organized logically to ensure future extensibility.